pallets/circuit/src/lib.rs

```diff
@@ -248,7 +268,36 @@ pub mod pallet {
         // dispatched.
         //
         // This function must return the weight consumed by `on_initialize` and `on_finalize`.
-        fn on_initialize(_n: T::BlockNumber) -> Weight {
+        fn on_initialize(n: T::BlockNumber) -> Weight {
+            // Check every XtxTimeoutCheckInterval blocks
+            if n % T::XtxTimeoutCheckInterval::get() == T::BlockNumber::from(0u8) {
+                // Go over all unfinished Xtx to find those that timed out
+                <ActiveXExecSignalsTimingLinks<T>>::iter()
+                    .find(|(_xtx_id, timeout_at)| {
+                        timeout_at <= &frame_system::Pallet::<T>::block_number()
+                    })
+                    .map(|(xtx_id, _timeout_at)| {
+                        let mut local_xtx_ctx = Self::setup(
+                            CircuitStatus::RevertedTimedOut,
+                            &Self::account_id(),
+                            Zero::zero(),
+                            Some(xtx_id),
+                        )
+                        .unwrap();
+
+                        local_xtx_ctx.xtx.status = CircuitStatus::RevertedTimedOut;
+                        Self::apply(&mut local_xtx_ctx, None).unwrap();
```

Comment:
naughty unwrap here

Reply:
yeah this is bad. I replaced it with a `kill` method on the following PR which shouldn't fail